netcdftime/_netcdftime.pyx

```diff
@@ -1151,8 +1156,209 @@ def time2index(times, nctime, calendar=None, select='exact'):
+        self.microsecond = microsecond
+        self._calendar = calendar
+        # Can't initialize _utime here: utime creates a datetime instance (origin).
+        # This avoids an unbounded recursion.
+        self._utime = None
+
+    @property
+    def format(self):
+        return '%Y-%m-%d %H:%M:%S'
+
+    def strftime(self, format=None):
+        if format is None:
+            format = self.format
+        return _strftime(self, format)
+
+    def replace(self, **kwargs):
+        "Return datetime with new specified fields."
+        args = {"year": self.year,
+                "month": self.month,
+                "day": self.day,
+                "hour": self.hour,
+                "minute": self.minute,
+                "second": self.second,
+                "microsecond": self.microsecond,
+                "dayofwk": self.dayofwk,
+                "dayofyr": self.dayofyr,
+                "calendar": self.calendar}
+
+        for name, value in kwargs.items():
+            args[name] = value
+
+        return datetime(**args)
+
+    def timetuple(self):
+        return (self.year, self.month, self.day, self.hour,
+                self.minute, self.second, self.dayofwk, self.dayofyr, -1)
+
+    def _to_real_datetime(self):
+        return real_datetime(self.year, self.month, self.day,
+                             self.hour, self.minute, self.second,
+                             self.microsecond)
+
+    def __repr__(self):
+        return self.strftime(self.format)
+
+    def __hash__(self):
+        try:
+            d = self._to_real_datetime()
+        except ValueError:
+            return hash(self.timetuple())
+        return hash(d)
+
+    def __richcmp__(self, other, int op):
+        if hasattr(other, 'strftime'):
+            self_str = self.strftime('%Y-%m-%d %H:%M:%S')
+            other_str = other.strftime('%Y-%m-%d %H:%M:%S')
+            return PyObject_RichCompare(self_str, other_str, op)
+        return NotImplemented
+
+    def __reduce__(self):
+        """special method that allows instance to be pickled"""
+        args = (self.year,self.month,self.day,self.hour,
+                self.minute,self.second,self.microsecond,
+                self.dayofwk,self.dayofyr)
+        return (self.__class__,args)
+
+    @property
+    def calendar(self):
+        return self._calendar
+
+    @property
+    def utime(self):
+        if self._utime is None:
+            self._utime = utime("seconds since 1-1-1", self._calendar)
+        return self._utime
+
+    def __add__(self, other):
+        if isinstance(self, datetime) and isinstance(other, timedelta):
+            date = self
+            delta = other
+            utime = date.utime
+        elif isinstance(self, timedelta) and isinstance(other, datetime):
+            date = other
+            delta = self
+            utime = date.utime
+        else:
+            raise TypeError("incompatible netcdftime.datetime.__add__() arguments")
```

Comment:
Just write `return NotImplemented` here. Python takes care of turning that into a `TypeError`.